src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -281,6 +282,49 @@ def with_src_position(
     return expr_ast
 
 
+def _fill_ast_with_column(
+    ast: proto.Expr, value: "snowflake.snowpark.Column"
+) -> None:
+    """Copy from a Column object's AST into an AST expression.
+
+    Args:
+        ast (proto.SpColumnExpr): A previously created Expr() or SpColumnExpr() IR entity intance to be filled
+        value (snowflake.snowpark.Column): The value from which to populate the provided ast parameter.
+
+    Raises:
+        NotImplementedError: Raised if the Column object does not have an AST set.
+    """
+    if value._ast is None and FAIL_ON_MISSING_AST:
+        raise NotImplementedError(
+            f"Column({value._expression})._ast is None due to the use of a Snowpark API which does not support AST logging yet."
+        )
+    elif value._ast is not None:
+        ast.CopyFrom(value._ast)
+
+
+def _fill_ast_with_snowpark_column_or_sql_expr(
```

Comment:
dito

Reply:
I was thinking of following your pattern with the renamed `build_const_from_python_val` and renaming the `ast` parameters to `expr_builder`. Does that sound reasonable? The functions would be called
- `build_expr_from_snowpark_column`
- `build_expr_from_snowpark_column_or_sql_str`

If this pattern is more informative we should probably also rename `_fill_ast_with_snowpark_column_or_literal` to `build_expr_from_snowpark_column_or_python_val`, (potentially also `build_const_from_python_val` to `build_expr_from_python_val`  and change the parameter name in `build_const_from_python_val` from `ast` to `expr_builder` as well)

Reply:
An additional note here is that the type hints defined at the bottom of `type_utils.py` provide a good indication of the "build" methods we'll need and could potentially guide the naming scheme as well.
```python
# Type hints
ColumnOrName = Union["snowflake.snowpark.column.Column", str]
ColumnOrLiteralStr = Union["snowflake.snowpark.column.Column", str]
ColumnOrSqlExpr = Union["snowflake.snowpark.column.Column", str]
LiteralType = Union[VALID_PYTHON_TYPES_FOR_LITERAL_VALUE]
ColumnOrLiteral = Union["snowflake.snowpark.column.Column", LiteralType]
```

Reply:
Yeah, these names sound great.